### DISCUSSION TITLE
Welcome to droneengage_camera Discussions!

### DISCUSSION THREAD

**Discussion by DroneEngage** [DroneEngage/droneengage_camera#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by magoulasGio** [comment]
HELP please! I have a raspberry pi 2 w connected with FC and everything is working perfectly BUT! I can only get photos from raspberry pi camera but no video! i dont  know how to fix it...

**Comment by HefnySco** [comment]
Hello,
Since you can get photos that means you have access to camera.
It seems that the video webrtc connection is not established.
Are you using your local wifi to run RPI?
Are you using ArduPilot webclient or a local server?
Thanks

**Reply by magoulasGio** [reply]
It doesn't matter if I use wifi or not the result is the same. I iam using webclient.

**Reply by magoulasGio** [reply]
Thank you so much for your reply!

**Reply by magoulasGio** [reply]
So you think that i have to edit de_comm jason file to fix it or the problem is somewhere else?

**Reply by HefnySco** [reply]
> It doesn't matter if I use wifi or not the result is the same. I iam using webclient.

- If you are using WIFI in your Home and Connect via https://cloud.ardupilot.org:8001/webclient.html using your home computer from the same LAN then you should see the video.

- sometimes problems happens when people try to connect from 4G and it seemds to me in some mobile networks it does not work.

I believe that your case  is the first one which means most probably there is something failing when you run camera module and start streaming.
you need to stop camera module using `sudo systemctl stop de_camera.service` then you need to manually go to `cd /home/pi/drone_engage/de_camera` and run `sudo de_camera64.so`

and then try video streaming and let me know what happens.

also please make sure that you are running Bullseye with legacy camera support which I am sure you do since you already capture photoes.

**Reply by magoulasGio** [reply]
Ok... I just did what you said... and it was keep giving the same message again and again...  
![Capture](https://github.com/DroneEngage/droneengage_camera/assets/171146170/6596f9c4-ae3d-4117-85db-74b35f4b6882)

**Reply by HefnySco** [reply]
Please note that /dev/video0 is the only valid video driver.
you can change config file to 
```

   //  "camera_start_index"        :0, 
   //  "camera_end_index"          :10, 
   //  "one_session_per_camera"    : true,
       "camera_list": [
         {"name": "MainCam", "device_num": 0} 
         
       ],
```

**Reply by HefnySco** [reply]
please select stream and let me know what is the log of the app as you sent before in an image.

**Reply by magoulasGio** [reply]
Ok. I edited config.module.json file. the changes where that i commented this (//  "one_session_per_camera"    : true,) and i changed the name of the camera. it changed also in webclient.  But nothing...   

Do i have to edit anything else? i have very little experience as you can tell... 
![Capture](https://github.com/DroneEngage/droneengage_camera/assets/171146170/48422ca6-f15c-4e81-893f-057238bb52ef)

**Comment by HefnySco** [comment]
I can do online test with you.
Please email me private and we can setup a meeting.
***@***.***

**Reply by magoulasGio** [reply]
Hello hefny. I tried emailing you but something probably went wrong.

**Reply by HefnySco** [reply]
m o h a m m a d . h e f n y @ g mail

**Comment by magoulasGio** [comment]
Yes tha is correct... Maybe my emails went to junk or something...



Sent from my Galaxy



-------- Original message --------
From: MHefny ***@***.***>
Date: 19/6/24 5:39 am (GMT+02:00)
To: DroneEngage/droneengage_camera ***@***.***>
Cc: magoulasGio ***@***.***>, Comment ***@***.***>
Subject: Re: [DroneEngage/droneengage_camera] Welcome to droneengage_camera Discussions! (Discussion #1)


m o h a m m a d . h e f n y @ g mail

**Comment by DroneEngage** [comment]
SOLUTION: Issue was a missing STUN server address for WEBRTC.
